src/StringUtils.h

```diff
@@ -223,7 +223,7 @@ bool StringToInteger(const AString & a_str, T & a_Num)
 /** Returns an integer from a key-value string map.
```

Comment:
Only digits and optional `-` and `+` are accepted (see `StringToInteger` above that it uses). If the input contains anything else, the parsing fails and `a_Default` is returned. That's why it has `Integer` in the name - it only accepts integers.